LLM Testing/Code-Testing/src/main/java/ShellScriptRunner.java

```diff
@@ -87,10 +73,33 @@ public static TestResultAnalyzer runTesting(String repositoryDirectory) {
 
         } else {
             System.out.println("Build Failed");
-            System.out.println(compileResponse);
             String errorMessages = TestResultAnalyzer.getCompilationErrors(compileResponse);
+            ChatGPTAPI app = new ChatGPTAPI();
+            try{
+                String promptTemplate = app.getFileFromResource("Prompt Templates/NaturalLanguageError.txt");
+                String responseContent = app.getFileFromResource("content.txt");
+                String newPrompt = new PromptWriter(promptTemplate, responseContent, errorMessages, "naturalLanguage")
+                        .createNaturalLanguageErrorPrompt();
+                String request = "src/main/java/NaturalLanguageRequest.json";
+                Request newRequest = new Request("gpt-3.5-turbo", newPrompt); // object for gson to convert
+                Gson gson = new GsonBuilder().setPrettyPrinting().disableHtmlEscaping().create();
+                JsonElement jsonElement = gson.toJsonTree(newRequest);
+                String jsonString = gson.toJson(jsonElement);
+                try(PrintWriter out = new PrintWriter(request)){
+                    out.println(jsonString);
+                }
+                try(PrintWriter out = new PrintWriter("src/main/resources/newNaturalLanguageErrorPrompt.txt")){
+                    out.println(newPrompt);
+                }
+                catch (IOException e){
+                    System.out.println(e);
+                }
+                ChatGPTAPI.sendNaturalLanguageErrorRequest(key);
+            }
+            catch (IOException e){
+                System.out.println(e);
+            }
```

Comment:
Not sure if we should have API calls within the ShellScriptRunner class if possible. But if we have to, it would be good to possibly have a single method call instead of this chunk of code.

Prob need to have a quick discussion about this.

Reply:
Done